### PR TITLE
Fixed for Ability to create Container Registry with zone redundancy

### PR DESCRIPTION
**Pull request by v-hongli1**
relate to [issue#38488](https://github.com/Azure/azure-sdk-for-java/issues/38488)

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by weidongxu-microsoft** [comment]
1. can it be disabled in update flow?
2. should we return `boolean` (and name it `isZoneRedundencyEnabled()`)?

@XiaofeiCao

**Comment by XiaofeiCao** [comment]
`isZoneRedundancyEnabled()` should be more appropriate.

I'm fine with both `enableZoneRedundancy` and `withZoneRedundancy`. If we consider zone redundancy as an ability to have, `withZoneRedundancy` makes sense too.

File: sdk/resourcemanager/azure-resourcemanager-containerregistry/src/main/java/com/azure/resourcemanager/containerregistry/implementation/RegistryImpl.java [hunk 1]
```diff
@@ -226,8 +226,8 @@ public boolean isDedicatedDataEndpointsEnabled() {
     }
 
     @Override
-    public ZoneRedundancy zoneRedundancy() {
-        return this.innerModel().zoneRedundancy();
+    public boolean isZoneRedundancyEnabled() {
+        return !Objects.isNull(this.innerModel().zoneRedundancy()) && ZoneRedundancy.ENABLED.equals(this.innerModel().zoneRedundancy());
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-containerregistry/src/main/java/com/azure/resourcemanager/containerregistry/implementation/RegistryImpl.java:230]: ResourceManagerUtils.toPrimitiveBoolean

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-containerregistry/src/main/java/com/azure/resourcemanager/containerregistry/implementation/RegistryImpl.java:230]: The input parameter of this static method `ResourceManagerUtils.toPrimitiveBoolean` must be of type `Boolean`, but the type of `ZoneRedundancy` is an enumeration. So I think `ResourceManagerUtils.toPrimitiveBoolean` is not applicable here.

**Comment by weidongxu-microsoft** [comment]
can it be disabled in update flow?

**Comment by v-hongli1** [comment]
@weidongxu-microsoft 
1. I tried updating the initial settings of "ZoneRedundancy", but the backend doesn't support it. At the same time, similar updates are not supported on the portal.
2. Now the `Update` process uses `updateParameters` to save update items and values, but the member object associated with `ZoneRedundancy` cannot be found in `updateParameters`, so the update settings of `ZoneRedundancy` cannot be recorded in `updateParameters`.

File: sdk/resourcemanager/azure-resourcemanager-containerregistry/src/main/java/com/azure/resourcemanager/containerregistry/implementation/RegistryImpl.java (hunk 1 above)
Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-containerregistry/src/main/java/com/azure/resourcemanager/containerregistry/implementation/RegistryImpl.java:230]: Seems the logic can be simplified to just `ZoneRedundancy.ENABLED.equals(this.innerModel().zoneRedundancy()`?

**Review by XiaofeiCao — APPROVED** [review]

File: sdk/resourcemanager/azure-resourcemanager-containerregistry/src/main/java/com/azure/resourcemanager/containerregistry/implementation/RegistryImpl.java (hunk 1 above)
Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-containerregistry/src/main/java/com/azure/resourcemanager/containerregistry/implementation/RegistryImpl.java:230]: The judgment of `isNull` is still needed.
image

Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-containerregistry/src/main/java/com/azure/resourcemanager/containerregistry/implementation/RegistryImpl.java:230]: When `zoneRedundancy` is null, it means `isZoneRedundancyEnabled` is false, right? Seems to align with `ZoneRedundancy.ENABLED.equals(this.innerModel().zoneRedundancy()`'s result.
Anyway, no hurt to have the null check.

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-containerregistry/src/main/java/com/azure/resourcemanager/containerregistry/implementation/RegistryImpl.java:230]: Fixed in the [pr#38592](https://github.com/Azure/azure-sdk-for-java/pull/38592)

**Pull request merged**